### DISCUSSION TITLE
Can't find any Sonos players -- Boost?

### DISCUSSION THREAD

**Discussion by ericbogs** [hklages/node-red-contrib-sonos-plus#130] [Q&A] [answered]
Hi there,

I'm unable to scan for or add any Sonos nodes to my sonos-config node.

My Node-RED instance is running on Synology NAS at 192.168.7.253. I have 6 Sonos players connect via a Sonos Boost Hub. The Boost is connect via wired ethernet to my hub, and given IP 192.168.7.169. My other Sonos players are primarily connected via Sonos mesh network (WM: 0), but have IP addresses (e.g. Bedroom Sonos is 192.168.7.245).

Here are things I've tried:

1. Entering the Boost IP (192.168.7.169) and scanning -> "Did not find any sonos any player"
2. Entering the Bedroom Sonos IP (192.168.7.245) and scanning -> "Did not find any sonos any player"
4. Entering the the Boost IP (192.168.7.169) and the Boost serial number (78-28-CA-77-XX-XX:9) and scanning -> "Did not find any sonos any player"
5. Entering the Bedroom Sonos IP (192.168.7.245) and the Bedroom Sonos serial number (78-28-CA-05-XX-XX:G) scanning -> "Did not find any sonos any player"
7. Deploying with Boost IP & Boost serial number -> "create and subscribe: couldn't find any player by serial :: Details: none"
8. Deploying with Bedroom Sonos IP & Bedroom Sonos serial number -> "create and subscribe: couldn't find any player by serial :: Details: none"

Ideas?

**Comment by hklages** [comment, marked as answer]
Hi. Try either
a) remove any data you entered  (serial, ip address) and then push the scan button 
or 
b) dont scan, just enter the ip address (and remove the serial number). 

Does that work?

**Reply by ericbogs** [reply]
Hi. Your (b) recommendation worked (just using the IP address). Thanks!

**Comment by evazzoler** [comment]
Hi, I have the same exact problem. Installed today the palette from scratch and restarted node red.
I own only Play:1. No way to find them trying all the combination here above.
Any suggestion?

**Comment by hklages** [comment]
@evazzoler You can use the Sonos app, systems look up the ip address of that player and enter it in the field. That always works

Right? 

The search may fail in case of special network / dns / environment. What is your environment?

**Reply by evazzoler** [reply]
Sure! I currently use it with the IP, but I'd prefer to use it with the serial number that will never change.
(anyway the IP is reserved with the MAC)

Not a problem for me, I only inform that a function is not working.

**Reply by hklages** [reply]
I suggest that you use DNS names. This package accepts DNS names. 

Serial numbers are not recommend because at every Node-RED deploy the package starts a discovery in the network to figure out the ip address of that serial number. Especially if you have a lot of players that slows down the deploy process (and if your network is poor it may run in a time out without discovering the ip address). 

Your opinion about DNS?

**Comment by hklages** [comment]
Thanks for letting me know the issue. 

I guess it is a "special docker" environment - right? Do you bridge ports?

**Reply by evazzoler** [reply]
No, no, it's a "bare metal" node red install on a Debian running on an industrial Raspberry Pi ("strato pi").

**Comment by hklages** [comment]
Are the Raspberry Pi and the Sonos Player in the Same Network? What IP
Adresse?

**Reply by evazzoler** [reply]
Node Red: 10.10.10.10/24
Sonos Play:1: 10.10.10.161/24

Multicast and broadcast is working on the raspberry because some services are working well (i.e.: knx over IP, Philips HUE, etc.)

**Comment by hklages** [comment]
I will come back to you on Monday, after having returned from my trip. I
need access to my players to further investigate.

**Comment by hklages** [comment]
@evazzoler 

The packages uses ssdp discovery on udp network , broadcast (M-SEARCH). It uses port 9000 and a randomly assigened port for listening. Both ports must be "open" for udp.

If you use a Docker container all ports must be mapped - best is to use same network (using bridge for randomly assigned ports is difficult). 

If Node-RED is running you can use the following requests from your Internet browser to initiate the discovery and test it. 
Let <ipandport> be the ip address including port of your Node-RED instance. 

`http://<ipandport>/nrcsp/discoverAllPlayerWithHost`
`http://<ipandport>/nrcsp/discoverAllPlayerWithSerialnumber`

Furthermore: You can set  the env variable to see all steps during the discovery. 
DEBUG = 'nrcsp:discovery, nrcsp:discovery-base'

Setting a ENV for Node-RED is described [here](https://discourse.nodered.org/t/how-to-set-env-variable/18612)

What is your outcome? Does it help?

**Reply by evazzoler** [reply]
Hi, unfortunately I don't get anything with the discovery.
Sever is under port 80 and /admin path, as you can see.
I tried to set the ENV following that post (editing settings.js) but no difference. I suppose I'm doing something wrong.
I don't use docker, node red is straight installed.

![immagine](https://github.com/hklages/node-red-contrib-sonos-plus/assets/9479876/0050e5a6-0ccd-4fab-be8b-5a159ad7d7bc)
![immagine](https://github.com/hklages/node-red-contrib-sonos-plus/assets/9479876/6aceabe8-2aaa-427c-b264-fc6635b95f6b)

**Comment by hklages** [comment]
@evazzoler "No, no, it's a "bare metal" node red install on a Debian running on an industrial Raspberry Pi ("strato pi")."

I just installed Node-RED with the standard script on my Raspberry Pi (Node-RED 3.1.7, NodeJS 18.19.1) and everything works fine. All my players are discoverd in a second. Also on my other test systesms (unraid with docker, Home Assistant, synology with docker, ...) it works.

So I assume, that there is something on your your "strato pi" or in your network blocking the access to your ports. Unfortunately I can not help you on that.
